webapp/channels/src/tests/react_testing_utils.tsx

```diff
@@ -22,87 +22,84 @@ import type {GlobalState} from 'types/store';
 export * from '@testing-library/react';
 export {userEvent};
 
-export const renderWithIntl = (component: React.ReactNode | React.ReactNodeArray, locale = 'en') => {
-    return render(<IntlProvider locale={locale}>{component}</IntlProvider>);
-};
-
-export const renderWithIntlAndStore = (component: React.ReactNode | React.ReactNodeArray, initialState: DeepPartial<GlobalState> = {}, locale = 'en', divContainer?: HTMLDivElement) => {
-    // We use a redux-mock-store store for testing, but we set up a real store to ensure the initial state is complete
-    const realStore = configureStore(initialState);
-
-    const store = mockStore(realStore.getState());
-
-    return render(
-        <IntlProvider locale={locale}>
-            <Provider store={store}>
-                {component}
-            </Provider>
-        </IntlProvider>,
-        {container: divContainer},
-    );
-};
+export type FullContextOptions = {
+    locale?: string;
+    useMockedStore?: boolean;
+}
+
+export const renderWithContext = (
```

Comment:
much simpler to use a single utility function rather than many of different variants